jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -479,7 +547,14 @@ ecma_concat_ecma_strings (ecma_string_t *string1_p, /**< first ecma-string */
     data_p = (lit_utf8_byte_t *) (long_string_desc_p + 1);
   }
 
-  string_desc_p->hash = lit_utf8_string_hash_combine (string1_p->hash, utf8_string2_p, utf8_string2_size);
+  lit_string_hash_t hash_start = string1_p->hash;
+
+  if (string1_is_uint32)
+  {
+    hash_start = lit_utf8_string_calc_hash (uint32_to_string_buffer1, utf8_string1_size);
```

Comment:
Couldn't be `hash_start = string1_p->u.uint32_number;` like in `ecma_init_ecma_string_from_uint32`?